core/extension/gdextension_manager.cpp

```diff
@@ -50,6 +50,11 @@ GDExtensionManager::LoadStatus GDExtensionManager::load_extension(const String &
 			extension->initialize_library(GDExtension::InitializationLevel(i));
 		}
 	}
+
+	for (const KeyValue<String, String> &kv : extension->class_icon_paths) {
+		gdextension_class_icon_paths[kv.key] = kv.value;
```

Comment:
Could possibly check in `gdextension.cpp` when we fill its internal map that keys match registered types? Two extensions shouldn't be able to register a class with the same name, so this would be sufficient, wdyt?

Reply:
It should not be able to register class with the same name, but can have icons for anything in the `.gdextension`, so I maybe it should only allow icons for the registered classes (but this is probably not trivial, since classes can be registered on different init levels).

Reply:
Yeah, I would just not support arbitrary icons, at least for now. I can see that checking against registered classes is indeed not trivial, though. Will need to look more into it, but I guess we can address it in a follow-up?

Reply:
For the time being I think it's ok if we just let whichever gets registered first or last take precedence. Naming conflicts should be rare and we can handle this later on.